dask/tests/test_order.py

```diff
@@ -351,7 +351,7 @@ def test_local_parents_of_reduction(abcde):
     b1, b2, b3 = [b + i for i in "123"]
     c1, c2, c3 = [c + i for i in "123"]
 
-    expected = [a3, a2, a1, b3, b2, b1, c3, c2, c1]
+    expected = [a3, b3, c3, a2, a1, b2, b1, c2, c1]
```

Comment:
What we had here previously was breadth first execution. This is actually counter to what we want or expect. As such it doesn't seem like this test was correct and we may have a bug in the synchronous scheduler currently, which this change has fixed. So we just fixed the test here to match that expected order